packages/server-wallet/src/__test__/sync-objectives.test.ts

```diff
@@ -0,0 +1,79 @@
+import {CreateChannelParams} from '@statechannels/client-api-schema';
+import Knex from 'knex';
+
+import {defaultTestConfig, overwriteConfigWithDatabaseConnection} from '../config';
+import {DBAdmin} from '../db-admin/db-admin';
+import {seedAlicesSigningWallet, seedBobsSigningWallet} from '../db/seeds/1_signing_wallet_seeds';
+import {DBObjective, ObjectiveModel} from '../models/objective';
+import {Wallet} from '../wallet';
+import {createChannelArgs} from '../wallet/__test__/fixtures/create-channel';
+import {bob} from '../wallet/__test__/fixtures/participants';
+
+import {getChannelResultFor, getPayloadFor} from './test-helpers';
+
+jest.setTimeout(10_000);
+const aWalletConfig = overwriteConfigWithDatabaseConnection(defaultTestConfig(), {
+  database: 'TEST_A',
+});
+const bWalletConfig = overwriteConfigWithDatabaseConnection(defaultTestConfig(), {
+  database: 'TEST_B',
+});
+
+let a: Wallet;
+let b: Wallet;
+
+beforeAll(async () => {
+  await Promise.all([DBAdmin.dropDatabase(aWalletConfig), DBAdmin.dropDatabase(bWalletConfig)]);
+  await DBAdmin.createDatabase(aWalletConfig);
+  await DBAdmin.createDatabase(bWalletConfig);
+  await Promise.all([
+    DBAdmin.migrateDatabase(aWalletConfig),
+    DBAdmin.migrateDatabase(bWalletConfig),
+  ]);
+  a = await Wallet.create(aWalletConfig);
+  b = await Wallet.create(bWalletConfig);
+
+  await seedAlicesSigningWallet(a.knex);
+  await seedBobsSigningWallet(b.knex);
+});
+afterAll(async () => {
+  await Promise.all([a.destroy(), b.destroy()]);
+  await Promise.all([DBAdmin.dropDatabase(aWalletConfig), DBAdmin.dropDatabase(bWalletConfig)]);
+});
+
+test('Objectives can be synced if a message is lost', async () => {
+  const createChannelParams: CreateChannelParams = createChannelArgs();
+
+  // We mimic not receiving a message containing objectives
+  const messageToLose = await a.createChannel(createChannelParams);
```

Comment:
[required] What's missing in this test suite is what happens when the message here is not lost, but `syncObjectives` is called.

It's possible that this is already resolved by some other test case. Is there a test that checks what happens when an objective is pushed multiple times?

Reply:
Updated in 44611f7